content/300-guides/400-migrate-to-prisma/03-migrate-from-mongoose.mdx

```diff
@@ -0,0 +1,1056 @@
+  title     String
+  content   String
+  published Boolean
+  v         Int     @map("__v")
+
+  author   users  @relation(fields: [authorId], references: [id])
+  authorId String @map("author") @db.ObjectId
+
+  categories   categories[] @relation(fields: [categoryIds], references: [id])
+  categoryIds String[]     @db.ObjectId
+}
+
+model users {
+  id      String        @id @default(auto()) @map("_id") @db.ObjectId
+  v       Int           @map("__v")
+  email   String        @unique(map: "email_1")
+  name    String
+  profile UsersProfile?
+  posts   posts[]
+}
+```
+
+</tab>
+
+</TabbedContent>
+
+### 2.5 Adjust the Prisma schema (optional)
+
+The models that were generated via introspection currently _exactly_ map to your database collections. In this section, you'll learn how you can adjust the naming of the Prisma models to adhere to [Prisma's naming conventions](/reference/api-reference/prisma-schema-reference#naming-conventions).
+
+Some of these adjustments are entirely optional and you are free to skip to the next step already if you don't want to adjust anything for now. You can go back and make the adjustments at any later point.
+
+As opposed to the current snake_case notation of Prisma models, Prisma's naming conventions are:
+
+- PascalCase for model names
+- camelCase for field names
+
+You can adjust the naming by _mapping_ the Prisma model and field names to the existing table and column names in the underlying database using `@@map` and `@map`.
+
+<Tip>
+
+You can use the [rename symbol](https://code.visualstudio.com/docs/editor/refactoring#_rename-symbol) operation to refactor model names by highlighting the model name, pressing <kbd>F2</kbd>, and finally typing the desired name. This will rename all instances where it is referenced and add the `@@map()` attribute to the existing model with its former name.
+
+</Tip>
+
+If your schema includes a [`versionKey`](https://mongoosejs.com/docs/guide.html#versionKey), change it to an optional field and add the `@ignore` attribute to the `v` field to exclude it from the generated Prisma Client. Marking the field as optional will allow you to create records in the collection.
+
+Also note that you can rename [relation fields](/concepts/components/prisma-schema/relations#relation-fields) to optimize the Prisma Client API that you'll use later to send queries to your database. For example, the `post` field on the `user` model is a _list_, so a better name for this field would be `posts` to indicate that it's plural.
+
+You can adjust update the `published` field by including the `@default` attribute to define the default value of the field.
+
+Here's an adjusted version of the Prisma schema that addresses these points:
+
+```prisma file=prisma/schema.prisma highlight=14,17,22,25,29,30,38,41,43,49;normal
+generator client {
+  provider = "prisma-client-js"
+}
+
+datasource db {
+  provider = "mongodb"
+  url      = env("DATABASE_URL")
+}
+
+type UsersProfile {
```

Comment:
Maybe rename this to `UserProfile`?

Reply:
This is the default name Mongoose assigns the embedded document.

This is the scenario where I ran into https://github.com/prisma/language-tools/issues/1122. I'm not sure if renaming the embedded document would affect existing embedded documents—worth exploring. I'll give it a try

Reply:
I would expect that the name of the embedded document is just a virtual construct. In the data set it is just defined by the property that contains it - and everything else is just for Mongoose or Prisma.